st2api/st2api/controllers/v1/packs.py

```diff
@@ -32,6 +55,128 @@
 
 LOG = logging.getLogger(__name__)
 
+ENTITIES = {
+    'action': (ActionsRegistrar, 'actions'),
+    'trigger': (TriggersRegistrar, 'triggers'),
+    'sensor': (SensorsRegistrar, 'sensors'),
+    'rule': (RulesRegistrar, 'rules'),
+    'alias': (AliasesRegistrar, 'aliases'),
+    'policy': (PolicyRegistrar, 'policy'),
+    'config': (ConfigsRegistrar, 'config')
+}
+
+
+class PackInitController(ActionExecutionsControllerMixin, RestController):
+
+    @jsexpose(body_cls=PackInitRequestAPI, status_code=http_client.ACCEPTED)
+    def post(self, args):
+        parameters = vars(args)
+
+        new_liveaction_api = LiveActionCreateAPI(action='packs.create',
+                                                 parameters=parameters,
+                                                 user=None)
+
+        execution = self._handle_schedule_execution(liveaction_api=new_liveaction_api)
+
+        return PackAsyncAPI(execution_id=execution.id)
+
+
+class PackInstallController(ActionExecutionsControllerMixin, RestController):
+
+    @jsexpose(body_cls=PackInstallRequestAPI, status_code=http_client.ACCEPTED)
+    def post(self, pack_install_request):
+        parameters = {
+            'packs': pack_install_request.packs
+        }
+
+        new_liveaction_api = LiveActionCreateAPI(action='packs.install',
+                                                 parameters=parameters,
+                                                 user=None)
+
+        execution = self._handle_schedule_execution(liveaction_api=new_liveaction_api)
+
+        return PackAsyncAPI(execution_id=execution.id)
+
+
+class PackUninstallController(ActionExecutionsControllerMixin, RestController):
+
+    @jsexpose(body_cls=PackInstallRequestAPI, arg_types=[str], status_code=http_client.ACCEPTED)
+    def post(self, pack_uninstall_request, ref_or_id=None):
+        if ref_or_id:
+            parameters = {
+                'packs': [ref_or_id]
+            }
+        else:
+            parameters = {
+                'packs': pack_uninstall_request.packs
+            }
+
+        new_liveaction_api = LiveActionCreateAPI(action='packs.uninstall',
+                                                 parameters=parameters,
+                                                 user=None)
+
+        execution = self._handle_schedule_execution(liveaction_api=new_liveaction_api)
+
+        return PackAsyncAPI(execution_id=execution.id)
+
+
+class PackRegisterController(RestController):
+
+    @jsexpose(body_cls=PackRegisterRequestAPI)
+    def post(self, pack_register_request):
+        if pack_register_request and hasattr(pack_register_request, 'types'):
+            types = pack_register_request.types
+        else:
+            types = ['runner', 'action', 'trigger', 'sensor', 'rule', 'rule_type', 'alias',
```

Comment:
As I said previously, I'd rather we have all our content types in form of a tree with explicit dependencies (like runner before rule and rule_type before rule).